tests/integration/targets/vmware_ops_provision_virtual_esxi_test/tasks/cleanup_esxi.yml

```diff
@@ -0,0 +1,64 @@
+- name: Power off ESXi VM
+  ansible.builtin.include_role:
+    name: cloud.vmware_ops.provision_vm
+  vars:
+    provision_vm_hostname: "{{ provision_virtual_esxi_hostname }}"
+    provision_vm_username: "{{ provision_virtual_esxi_username }}"
+    provision_vm_password: "{{ provision_virtual_esxi_password }}"
+    provision_vm_validate_certs: false
+    provision_vm_port: "{{ provision_virtual_esxi_port }}"
+    provision_vm_cluster: "{{ provision_virtual_esxi_cluster }}"
+    provision_vm_datacenter: "{{ provision_virtual_esxi_datacenter }}"
+    provision_vm_name: "{{ provision_virtual_esxi_vms[0].name }}"
+    provision_vm_state: "poweredoff"
+
+- name: Cleanup Virtual Esxi
+  ansible.builtin.include_role:
+    name: cloud.vmware_ops.provision_vm
+  vars:
+    provision_vm_hostname: "{{ provision_virtual_esxi_hostname }}"
+    provision_vm_username: "{{ provision_virtual_esxi_username }}"
+    provision_vm_password: "{{ provision_virtual_esxi_password }}"
+    provision_vm_validate_certs: false
+    provision_vm_port: "{{ provision_virtual_esxi_port }}"
+    provision_vm_cluster: "{{ provision_virtual_esxi_cluster }}"
+    provision_vm_datacenter: "{{ provision_virtual_esxi_datacenter }}"
+    provision_vm_name: "{{ provision_virtual_esxi_vms[0].name }}"
+    provision_vm_state: "absent"
+    provision_vm_force: true
+
+- name: Check if ESXi VM is down
+  command: ping -c 1 -W 5 {{ esxi_host.instance.ipv4 }}
+  register: ping_check_result
+  ignore_errors: true
+
+- name: Set fact if ping is not available
+  set_fact:
+    ping_not_available: "{{ ping_check_result.rc != 0 }}"
+
+- name: Verify VM absence in vCenter
+  community.vmware.vmware_guest_info:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    datacenter: "{{ provision_virtual_esxi_datacenter }}"
+    validate_certs: no
+    name: "{{ provision_virtual_esxi_vms[0].name }}"
+  register: vm_info
+  ignore_errors: true
+
+- name: Fail if VM is still present in vCenter
+  fail:
+    msg: "The VM {{ provision_virtual_esxi_vms[0].name }} is still present in vCenter."
+  when: vm_info.instance is defined
+
+- name: Cleanup Resource Pool
+  community.vmware.vmware_resource_pool:
+    hostname: "{{ provision_virtual_esxi_hostname }}"
+    username: "{{ provision_virtual_esxi_username }}"
+    password: "{{ provision_virtual_esxi_password }}"
+    validate_certs: "{{ provision_virtual_esxi_validate_certs }}"
+    datacenter: "{{ provision_virtual_esxi_datacenter }}"
+    cluster: "{{ provision_virtual_esxi_cluster }}"
+    resource_pool: "{{ resource_pool_name }}"
+    state: absent
```

Comment:
what would happen if we directly delete the resource pool, without deleting first the esxi VM? 
if deleting the resource pool removes also the esxi host, then we could just run this task and drop the previous tasks?

Reply:
for instance [here](https://github.com/redhat-cop/cloud.vmware_ops/pull/64/files#diff-fad1ce2dd2617b5d75f1b3d62a8a42347cda2ff173e9dcc31ddec8172d2b72c7R130-R140) we will just run provision_vm role to remove the esxi host

Reply:
hmm for me it did not deleted the resource pool when I added this and deleted the manual removal:
```
- name: Cleanup Virtual Esxi
  ansible.builtin.include_role:
    name: cloud.vmware_ops.provision_vm
  vars:
    provision_vm_hostname: "{{ provision_virtual_esxi_hostname }}"
    provision_vm_username: "{{ provision_virtual_esxi_username }}"
    provision_vm_password: "{{ provision_virtual_esxi_password }}"
    provision_vm_validate_certs: false
    provision_vm_port: "{{ provision_virtual_esxi_port }}"
    provision_vm_cluster: "{{ provision_virtual_esxi_cluster }}"
    provision_vm_datacenter: "{{ provision_virtual_esxi_datacenter }}"
    provision_vm_name: "{{ provision_virtual_esxi_vms[0].name }}"
    provision_vm_resource_pool: "{{ resource_pool }}"
    provision_vm_state: "absent"
    provision_vm_force: true
```

Reply:
I meant what happens if we only delete the resource pool with `Cleanup Resource Pool` task:

```
- name: Cleanup Resource Pool
  community.vmware.vmware_resource_pool:
    hostname: "{{ provision_virtual_esxi_hostname }}"
    username: "{{ provision_virtual_esxi_username }}"
    password: "{{ provision_virtual_esxi_password }}"
    validate_certs: "{{ provision_virtual_esxi_validate_certs }}"
    datacenter: "{{ provision_virtual_esxi_datacenter }}"
    cluster: "{{ provision_virtual_esxi_cluster }}"
    resource_pool: "{{ resource_pool_name }}"
    state: absent
```
does the VM still exists in the vcenter?

Reply:
ah, checking

Reply:
OK I checked - when deleting the resource pool without powering off or removing the vm, the resource pool is deleted but the vm is moving to the cluster as a stand alone

Reply:
Iv'e merged it for now as the checks are passing, I'll open another one for fixes if needed